### PR TITLE
New context menus added for different regions of the DataGrid

### LINKED ISSUE #123: UI/UX for User Story: Using a context menu for basic data cleaning
## Description
This is an example of using context menus to do some basic data cleaning such as deleting irrelevant parts of data. The gif also displays how you could use context menus to add new rows and columns.

context-menu

Here's an image of the different context menus depending on what you click:
context-menus

## Related Issues
None

## Questions

Should the context menu be simplified more (fewer commands)?

Are there any commands you would add to the context menus?

Any questions on the wording of the commands or what a command does?

Thoughts and feedback are greatly appreciated!

## Interactive Prototype
[Prototype Link](https://www.figma.com/proto/SLJlzW2DjPPDTTbV64A8bN/Shrimp-n-Grids?node-id=534%3A1984&scaling=min-zoom)

### LINKED ISSUE #126: Apply changes to the context menues
**Describe the new feature being added**
There should be different context menus according to the click region.

**How will this feature improve this extension**
Give users of the editor a more intuitive experience.

### LINKED ISSUE #131: Weird behavior of remove column when the first column is blank
**Actual behavior**
,1,2,3
,4,5,6

gets mapped to 

D

,2,3
,5,6

when removing the [1 4] column.

### PR DESCRIPTION
**Pull request by ghost**
# Description of the Pull Request
Some of the context menu changes requested by @ryuntalan are now in place.

Designs can be found in #123 

### Issue being fixed:
Fixes #126
Fixes #131 

### Changes proposed:

- DOM children added to the viewport (one for the body region, one for the row headers, and one for the column headers)
- A signal is passed from `handler` to `widget` when these DOM elements need to be updated
- 3 different context menus now defined in `index.ts`.
- Commands added for inserting columns/rows in front or behind.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -137,9 +137,21 @@ function addCommands(
...
   const { commands } = app;
-  const SELECTOR = '.jp-CSVViewer-grid';
+  const GLOBAL_SELECTOR = '.jp-CSVViewer-grid';
+  const BODY_SELECTOR = '.jp-background';
+  const COLUMN_HEADER_SELECTOR = '.jp-column-header';
+  const ROW_HEADER_SELECTOR = '.jp-row-header';
 
-  commands.addCommand(CommandIDs.addRow, {
+  commands.addCommand(CommandIDs.insertRowAbove, {
+    label: 'Add Row',
```
Comment (kgoo124) [inline src/index.ts:146]: Insert Row Above

File: src/index.ts [hunk 2]
```diff
@@ -137,9 +137,21 @@ function addCommands(
...
+    execute: () => {
+      // emit a signal to the EditableDSVModel
+      tracker.currentWidget &&
+        tracker.currentWidget.content.changeModelSignal.emit('add-row');
```
Comment (kgoo124) [inline src/index.ts:150]: Should be `insert-row-above`

File: src/index.ts [hunk 3]
```diff
@@ -137,9 +137,21 @@ function addCommands(
...
+    }
+  });
+
+  commands.addCommand(CommandIDs.insertRowBelow, {
     label: 'Add Row',
```
Comment (kgoo124) [inline src/index.ts:155]: Insert Row Below here and for the signal emitted

**Review by thegooman — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 3 above)
Comment (lmcnichols) [reply src/index.ts]: Changes made!

**Comment by kgoo124** [comment]
Inserting a column (either direction) and then removing any column results in the data being incorrectly displayed

**Review by thegooman — APPROVED** [review]
Bug fixed! Great work!

**Pull request merged**